barretenberg/cpp/src/barretenberg/dsl/acir_format/ec_operations.cpp

```diff
@@ -13,19 +14,126 @@
+        if (!input.input1_infinite.is_constant || (!input.input1_x.is_constant && !input.input1_y.is_constant)) {
+            input1_point = to_grumpkin_point(
+                input.input1_x, input.input1_y, input.input1_infinite, has_valid_witness_assignments, true, builder);
+        } else {
+            // If not, the coordinates are mixed constant/witness, and we generate witness so that the point is using
+            // only witnesses.
+            input1_point = to_witness_grumpkin_point(
+                input.input1_x, input.input1_y, input.input1_infinite, has_valid_witness_assignments, true, builder);
+        }
+        result = input1_point.dbl();
+    } else {
+        // Regular addition
+        // if one point is (constant) zero, we simply return the other point.
+        if (input.input2_infinite.is_constant && input.input1_infinite.is_constant) {
+            if (get_value(input.input1_infinite, builder) == 1) {
+                // input1 is infinity, so we can just return input2
+                result = to_grumpkin_point(input.input2_x,
+                                           input.input2_y,
+                                           input.input2_infinite,
+                                           has_valid_witness_assignments,
+                                           false,
+                                           builder);
+
+            } else if (get_value(input.input2_infinite, builder) == 1) {
+                // input2 is infinity, so we can just return input1
+                result = to_grumpkin_point(input.input1_x,
+                                           input.input1_y,
+                                           input.input1_infinite,
+                                           has_valid_witness_assignments,
+                                           true,
+                                           builder);
+            } else {
+
+                cycle_group_ct input1_point;
+                cycle_group_ct input2_point;
+                // all or nothing: the inputs must be all constant or all witness. Cf #1108 for more details.
+                if (!input.input1_x.is_constant || !input.input1_y.is_constant || !input.input1_infinite.is_constant ||
+                    !input.input2_x.is_constant || !input.input2_y.is_constant || !input.input2_infinite.is_constant) {
+                    // One of the input is a witness, so we ensure that all inputs are witness, by creating witness for
+                    // constant values.
+                    input1_point = to_witness_grumpkin_point(input.input1_x,
+                                                             input.input1_y,
+                                                             input.input1_infinite,
+                                                             has_valid_witness_assignments,
+                                                             true,
```

Comment:
Same here. Why is 1 true and 2 false?

Reply:
Here it matters, because the two points must be distinct (else it would be a doubling).